src/core/server/csp/csp_config.ts

```diff
@@ -48,30 +49,50 @@ export interface ICspConfig {
    * in a `Content-Security-Policy` header.
    */
   readonly header: string;
+
+  /**
+   * Flag indicating that the configuraion changes the csp
+   * rules from the defaults
+   */
+  readonly headerChangedFromDefault: boolean;
 }
 
 /**
  * CSP configuration for use in Kibana.
  * @public
  */
 export class CspConfig implements ICspConfig {
-  static readonly DEFAULT = new CspConfig();
-
   public readonly rules: string[];
   public readonly strict: boolean;
   public readonly warnLegacyBrowsers: boolean;
   public readonly header: string;
+  public readonly headerChangedFromDefault: boolean;
```

Comment:
NIT: header is a direct concat of rules, should probably be named after rules instead of header.

Reply:
That's understandable, I was conflicted because early on in development of this feature I feel into a trap where I hard modified the rules but headers didn't reflect the modification, so I tried to make it clear that this was based on the headers and also tried to remove the rules (it's unused) but it's somehow coupled with the input so I wasn't able to get that done.